### PR TITLE
feat: Enable remote invoke command

### PR DESCRIPTION
**Pull request by hnnasit**
#### Why is this change necessary?
This PR enables the remote invoke command and adds the related integration tests.

#### What side effects does this change have?
N/A

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [x] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/lib/cli_validation/remote_invoke_options_validations.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def wrapped(*args, **kwargs):
 
         # if no event nor event_file arguments are given, read from stdin
         if not event and not event_file:
-            LOG.debug("Neither --event nor --event-file options have been provided, reading from stdin")
+            LOG.info("Neither --event nor --event-file options have been provided, reading from stdin")
             kwargs["event_file"] = cast(TextIOWrapper, sys.stdin)
```
Comment (mndeveci) [inline samcli/lib/cli_validation/remote_invoke_options_validations.py:54]: I thought we decided to set this to empty event to align with `sam local invoke`?

File: tests/integration/remote/invoke/test_lambda_invoke.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        cls.stack_resources = {
+            resource_full_path: stack_resource_summary.physical_resource_id
+            for resource_full_path, stack_resource_summary in stack_resource_summaries.items()
+        }
+        cls.cfn_client = get_boto_client_provider_with_config()("cloudformation")
+        cls.lambda_client = get_boto_client_provider_with_config()("lambda")
+
+    def test_invoke_with_event_and_event_file_provided(self):
+        event_file_path = str(self.events_folder_path.joinpath("default_event.json"))
+        command_list = RemoteInvokeIntegBase.get_command_list(
```
Comment (mndeveci) [inline tests/integration/remote/invoke/test_lambda_invoke.py:45]: nit: wouldn't `self.get_command_list` work here?

File: tests/integration/testdata/remote_invoke/template-multiple-resources.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  RaiseExceptionFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: main.raise_exception
+      Runtime: python3.9
+      CodeUri: ./lambda-fns
+      Timeout: 600
+        
+# UTF-8 Test 😐
```
Comment (mndeveci) [inline tests/integration/testdata/remote_invoke/template-multiple-resources.yaml:57]: nit: do we need this here?

File: samcli/lib/cli_validation/remote_invoke_options_validations.py (hunk 1 above)
Comment (hnnasit) [reply samcli/lib/cli_validation/remote_invoke_options_validations.py:54]: Yes those changes are in the UX PR. Once that gets merged, I will resolve the conflicts in this PR.

File: tests/integration/testdata/remote_invoke/template-multiple-resources.yaml (hunk 3 above)
Comment (hnnasit) [reply tests/integration/testdata/remote_invoke/template-multiple-resources.yaml:57]: Nope, will remove it.

File: tests/integration/remote/invoke/test_lambda_invoke.py (hunk 2 above)
Comment (hnnasit) [reply tests/integration/remote/invoke/test_lambda_invoke.py:45]: Yup that should work as well, can change that everywhere.

**Pull request closed without merging**